core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseAsyncClientHandler.java

```diff
@@ -149,6 +198,8 @@ protected BaseAsyncClientHandler(SdkClientConfiguration clientConfiguration,
             runAndLogError(
                 log.logger(),
                 "Error thrown from TransformingAsyncResponseHandler#onError, ignoring.",
+                // TODO: Have the StandardAsyncHttpResponseHandler forward errors and create a new wrapper in the
+                // MakeAsyncHttpRequestStage to stop errors from retries escaping. Then remove this hack.
```

Comment:
Just curious, why is this a hack?

Reply:
Rereading it, I don't think this code is doing what I thought it was doing. I'm removing the TODO.